apps/remote_control/lib/lexical/remote_control/search/indexer/extractors/module.ex

```diff
@@ -0,0 +1,175 @@
+defmodule Lexical.RemoteControl.Search.Indexer.Extractors.Module do
+  @moduledoc """
+  Extracts module references and definitions from AST
+  """
+
+  alias Lexical.Ast
+  alias Lexical.Document.Position
+  alias Lexical.Document.Range
+  alias Lexical.ProcessCache
+  alias Lexical.RemoteControl.Search.Indexer.Entry
+  alias Lexical.RemoteControl.Search.Indexer.Metadata
+  alias Lexical.RemoteControl.Search.Indexer.Source.Block
+  alias Lexical.RemoteControl.Search.Indexer.Source.Reducer
+
+  # extract a module definition
+  def extract(
+        {:defmodule, defmodule_meta,
+         [{:__aliases__, module_name_meta, module_name}, module_block]},
+        %Reducer{} = reducer
+      ) do
+    %Block{} = block = Reducer.current_block(reducer)
+    aliased_module = resolve_alias(reducer, module_name)
+    module_position = Metadata.position(module_name_meta)
+    range = to_range(module_name, module_position)
+
+    entry =
+      Entry.definition(
+        reducer.document.path,
+        block.ref,
+        block.parent_ref,
+        aliased_module,
+        :module,
+        range,
+        Application.get_application(aliased_module)
+      )
+
+    module_name_meta = Reducer.skip(module_name_meta)
+
+    elem =
+      {:defmodule, defmodule_meta, [{:__aliases__, module_name_meta, module_name}, module_block]}
+
+    {:ok, entry, elem}
+  end
+
+  # This matches an elixir module reference
+  def extract({:__aliases__, metadata, maybe_module}, %Reducer{} = reducer)
+      when is_list(maybe_module) do
+    case module(reducer, maybe_module) do
+      {:ok, module} ->
+        start = Metadata.position(metadata)
+        range = to_range(maybe_module, start)
+        %Block{} = current_block = Reducer.current_block(reducer)
+
+        entry =
+          Entry.reference(
+            reducer.document.path,
+            make_ref(),
+            current_block.ref,
+            module,
+            :module,
+            range,
+            Application.get_application(module)
+          )
+
+        {:ok, entry}
+
+      _ ->
+        :ignored
+    end
+  end
+
+  # This matches an erlang module, which is just an atom
+  def extract(atom_literal, %Reducer{} = reducer) when is_atom(atom_literal) do
```

Comment:
errm, I don't think this is the case; When I prewalk the AST, I see stuff like: 
```elixir
:__block__, [line: 1, column: 21], [:do]}
```
which isn't an atom.

Reply:
It is an atom though, because `do` desugars into a keyword list where the key is `:do`. 

```elixir
if true do
  :ok
end

if(true, [{:do, :ok}])
```

Reply:
yes, but it isn't an atom in the sense anyone would ever care to find all references to "do"

Reply:
unless they want to stress the fuzzy finder.

Reply:
though with the module check, it won't be extracted.